### DISCUSSION TITLE
Does hooks need initConnection?

### DISCUSSION THREAD

**Discussion by jeremyhalin** [hyochan/react-native-iap#1574] [Q&A]
I'm experiencing errors like `Not initialized, Please call initConnection()` whereas I use hooks and have version 7.
I have a lot of errors in production, but absolutely none on an emulator or on my device in debug mode.
I don't really understand where all those errors are coming from.

Anyone using hooks and having zero error?

**Comment by jeremyhalin** [comment]
I'm bumping my question. I'm still getting A LOT of errors about `Not initialized, Please call initConnection()`.
Do I need to use `connected` from hooks to know if I'm connected or not before doing anything?
What happens if `connected` is false? How to connect again? Does it happen automatically?
To be honest, I'm thinking about going back to class component just for the shop part of my app because it was working well this way. Since I switch to hooks, I have so many errors.

**Reply by Arjan-Zuidema** [reply]
I don't think you need to call initConnection when using hooks. You should probably only use the functions and variables that are exposed when using `useIAP()`

**Reply by jeremyhalin** [reply]
I don't get the point of `connected` in hooks then.

**Reply by Arjan-Zuidema** [reply]
`connected` seem to be a state variable to check if the module is connected. I would guess it's something like this

```javascript
const [connected, setConnected] = React.useState(false)

React.useEffect(() => {
  initConnection().then(() => {
    setConnected(true);
  });

  return () => {
    endConnection()
  }
}, [])
```

**Reply by Arjan-Zuidema** [reply]
Also, I think you cannot use `getSubscriptions`/`getProducts` before `connected` === `true`

**Reply by SebastianMieszczanczyk** [reply]
You don't need to initialize connections on your own when you use the hook. As it was said above - use only methods that are exposed from the hook. It works correctly for me. `Connected` is just a state variable, so you can just use:
```
React.useEffect(() => {
if (!connected) {
   // refresh to initialize
   return;
}
// here you can do your steps after connection is established
}, [connected])
```

**Reply by SebastianMieszczanczyk** [reply]
<img width="861" alt="image" src="https://user-images.githubusercontent.com/53857792/160429934-713b5e0e-3191-4484-954a-cf87466aaca1.png">

So relying on code as on Promise (code in the answer above) is not correct.
